### PR TITLE
refactor: Add rule `no-constant-binary-expression` (no-changelog)

### PR DESCRIPTION
**Pull request by ivov**
https://eslint.org/docs/latest/rules/no-constant-binary-expression

### CODE REVIEW

File: packages/editor-ui/src/views/NodeView.vue [hunk 1]
```diff
@@ -2641,6 +2641,7 @@ export default defineComponent({
 				}
 
 				if (
+					// eslint-disable-next-line no-constant-binary-expression
```
Comment (ivov) [inline packages/editor-ui/src/views/NodeView.vue:2644]: Not quite sure here so I'd rather skip.

File: packages/nodes-base/nodes/HaloPSA/GenericFunctions.ts [hunk 2]
```diff
@@ -96,7 +96,7 @@ export async function haloPSAApiRequest(
 		return result;
 	} catch (error) {
 		const message = (error as JsonObject).message as string;
-		if (method === 'DELETE' || 'GET' || ('UPDATE' && message)) {
+		if (method === 'DELETE' || method === 'GET' || (method === 'UPDATE' && message)) {
```
Comment (netroy) [inline packages/nodes-base/nodes/HaloPSA/GenericFunctions.ts:99]: I'm guessing this was never actually tested, as this is equivalent of  `if (method === 'DELETE') {` 😅

**Review by netroy — APPROVED** [review]
Thanks 🙏🏽

**Review by netroy — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.17.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.17.0)

**Pull request merged**